File: src/main/java/com/d20charactersheet/framework/dac/dao/sql/rowmapper/KnownSpellsTableRowMapper.java
package com.d20charactersheet.framework.dac.dao.sql.rowmapper;

import java.sql.SQLException;

import org.jetbrains.annotations.NotNull;

import com.d20charactersheet.framework.boc.model.KnownSpellsTable;
import com.d20charactersheet.framework.dac.dao.sql.DataRow;
import com.d20charactersheet.framework.dac.dao.sql.RowMapper;

/**
 * Maps raw data to known spells table.
 */
public class KnownSpellsTableRowMapper implements RowMapper {

    @NotNull
    @Override
    public Object mapRow(@NotNull DataRow dataRow) throws SQLException {
        final KnownSpellsTable knownSpellsTable = new KnownSpellsTable();
        knownSpellsTable.setId(dataRow.getInt(0));
        knownSpellsTable.setName(dataRow.getString(1));
        return knownSpellsTable;
    }

}

File: src/main/java/com/d20charactersheet/framework/dac/dao/sql/rowmapper/KnownSpellsTableRowMapper.kt
package com.d20charactersheet.framework.dac.dao.sql.rowmapper

import com.d20charactersheet.framework.boc.model.KnownSpellsTable
import com.d20charactersheet.framework.dac.dao.sql.DataRow
import com.d20charactersheet.framework.dac.dao.sql.RowMapper
import java.sql.SQLException

/**
 * Maps raw data to known spells table.
 */
class KnownSpellsTableRowMapper : RowMapper {

    @Throws(SQLException::class)
    override fun mapRow(dataRow: DataRow): Any {
        val knownSpellsTable = KnownSpellsTable()
        knownSpellsTable.id = dataRow.getInt(0)
        knownSpellsTable.name = dataRow.getString(1)
        return knownSpellsTable
    }
}